test/rekt/features/broker/feature.go

```diff
@@ -37,6 +42,151 @@ import (
+	eventToSend := cloudevents.NewEvent()
+	eventToSend.SetID(uuid.New().String())
+	eventToSend.SetType(eventType)
+	eventToSend.SetSource(eventSource)
+	eventToSend.SetData(cloudevents.ApplicationJSON, []byte(eventBody))
+
+	//Install the broker
+	brokerName := feature.MakeRandomK8sName("broker")
+	f.Setup("install broker", broker.Install(brokerName, broker.WithEnvConfig()...))
+	f.Requirement("broker is ready", broker.IsReady(brokerName))
+	f.Requirement("broker is addressable", broker.IsAddressable(brokerName))
+
+	f.Setup("install sink1", eventshub.Install(sink1,
+		eventshub.ReplyWithTransformedEvent(transformedEventType, transformedEventSource, transformedBody),
+		eventshub.StartReceiver),
+	)
+	f.Setup("install sink2", eventshub.Install(sink2, eventshub.StartReceiver))
+	f.Setup("install sink3", eventshub.Install(sink3, eventshub.StartReceiver))
+
+	// filter1 filters the original events
+	filter1 := eventingv1.TriggerFilterAttributes{
+		"type":   eventType,
+		"source": eventSource,
+	}
+	// filter2 filters all events
+	filter2 := eventingv1.TriggerFilterAttributes{
+		"type": eventingv1.TriggerAnyFilter,
+	}
+	// filter3 filters events after transformation
+	filter3 := eventingv1.TriggerFilterAttributes{
+		"type":   transformedEventType,
+		"source": transformedEventSource,
+	}
+
+	// Install the trigger1 point to Broker and transform the original events to new events
+	f.Setup("install trigger1", trigger.Install(
+		trigger1,
+		brokerName,
+		trigger.WithFilter(filter1),
+		trigger.WithSubscriber(svc.AsKReference(sink1), ""),
+	))
+	f.Setup("trigger1 goes ready", trigger.IsReady(trigger1))
+	// Install the trigger2 point to Broker to filter all the events
+	f.Setup("install trigger2", trigger.Install(
+		trigger2,
+		brokerName,
+		trigger.WithFilter(filter2),
+		trigger.WithSubscriber(svc.AsKReference(sink2), ""),
+	))
+	f.Setup("trigger2 goes ready", trigger.IsReady(trigger2))
+
+	// Install the channel and corresponding subscription point to sink3
+	channelName := feature.MakeRandomK8sName("channel")
+	f.Setup("install channel", channel.Install(channelName,
+		channel.WithTemplate(),
+	))
+	sub := feature.MakeRandomK8sName("subscription")
+	f.Setup("install subscription", subscription.Install(sub,
+		subscription.WithChannel(channel.AsRef(channelName)),
+		createSubscriberFn(svc.AsKReference(sink3), ""),
+	))
+	f.Setup("subscription is ready", subscription.IsReady(sub))
+	f.Setup("channel is ready", channel.IsReady(channelName))
+
+	// Install the trigger3 point to Broker to filter the events after transformation point to channel
+	f.Setup("install trigger3", trigger.Install(
+		trigger3,
+		brokerName,
+		trigger.WithFilter(filter3),
+		trigger.WithSubscriber(channel.AsRef(channelName), ""),
+	))
+	f.Setup("trigger3 goes ready", trigger.IsReady(trigger3))
+
+	f.Requirement("install source", eventshub.Install(
+		source,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(eventToSend),
+	))
```

Comment:
We should install the source at any next phase where we verify "broker is ready" so we should put "broker is ready" and "broker is addressable" to the `Setup` phase (line 94-95) because "Requirement" steps could run in parallel